app/src/services/attestation.tsx

```diff
@@ -158,6 +210,87 @@ export const AttestationProvider: React.FC<AttestationProviderParams> = ({ child
     }
   }
 
+  const handleProofs = async (proof: ProofExchangeRecord, agent: BifoldAgent): Promise<void> => {
+    try {
+      // 0. if the proof is just now being requested, handle. Otherwise do nothing
+      if (proof.state === ProofState.RequestReceived) {
```

Comment:
How do you feel about using an "early return"? Rather than wrapping all the code in an "if" and indenting, just return if the condition is not met:

```javascript
if (proof.state != ProofState.RequestReceived) {
   return
}
```

I find they also help avoid Christmas trees of logic which can be error prone hand hard to follow:

```text
x
   x
      x
         x
      x
   x
x
```